docs/src/charmil_config.md

```diff
@@ -0,0 +1,170 @@
+# Charmil Config
+
+The Charmil Config package offers a convenient mechanism for both host and plugin developers to manage configurations in their command-line interface (CLI) applications made using [Cobra](https://github.com/spf13/cobra).
+
+## Features:
+
+- Uses [Viper](https://github.com/spf13/viper) under the hood.
+- Helps in maintaining all available configurations in a single central-local config file.
+- Provides the plugin developers with a set of methods to maintain a configuration map and export it to the host CLI.
+- Provides the host CLI developers with a set of methods to import plugin configuration maps and read/write configurations from/to a local config file.
+
+## Steps to Use:
+
+- ### For host CLI developers:
```

Comment:
We need to describe architecture.in separate.doc:

- what is host cli
- what is plugin

Separate issue/PR please